src/main/java/vn/triplet/controller/admin/UserController.java

```diff
@@ -1,14 +1,97 @@
 package vn.triplet.controller.admin;
 
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import vn.triplet.model.User;
+import vn.triplet.model.User.Role;
+import vn.triplet.service.UserService;
 
 @Controller
+@RequestMapping("/admin")
 public class UserController {
 
-	@RequestMapping("/admin/users")
-	public String index() {
+	@Autowired
+	UserService userService;
+
+	@GetMapping("/users")
+	public String index(Model model) {
+		List<User> users = userService.loadUsers();
+		model.addAttribute("users", users);
 		return "views/admin/user/users";
 	}
 
+	@GetMapping("/users/{id}")
+	public String show(@PathVariable("id") int id, Model model) {
+		User user = userService.findById(id);
+		if (user != null) {
+			model.addAttribute("user", user);
+			return "views/admin/user/user";
+		} else
+			return "views/admin/error/404";
+	}
+
+	@GetMapping(value = "/users/{id}/delete")
+	public String deleteUser(@PathVariable("id") Integer id, final RedirectAttributes redirectAttributes) {
+		User user = userService.findById(id);
+		if (user != null) {
+			if (userService.deleteUser(id)) {
+				redirectAttributes.addFlashAttribute("css", "success");
+				redirectAttributes.addFlashAttribute("msg", "User is deleted!");
+			} else {
+				redirectAttributes.addFlashAttribute("css", "error");
+				redirectAttributes.addFlashAttribute("msg", "Fail to delete user!");
+			}
+		} else {
+			redirectAttributes.addFlashAttribute("css", "error");
+			redirectAttributes.addFlashAttribute("msg", "User not found!");
```

Comment:
Code này lặp quá nè

suggest của chị là em định nghĩa 2 biến typeCss và message.
việc của em là check if else để setting giá trị cho 2 biến này, và sau cùng thì add vào atttribute để redirect thôi. sẽ clear và dễ đọc code hơn 
Ngoài ra, mục tiêu của chị là trả về kết quả sớm nhất cho method nếu có thể

String typeCss =  "error";
String message = "User not found!";
if (user == null) {
			redirectAttributes.addFlashAttribute("css", typeCss);
			redirectAttributes.addFlashAttribute("msg", message);
			return "redirect:/admin/users";
}


typeCss =  "error";
message = "Fail to delete user!"
if (userService.deleteUser(id)) {
		typeCss =  "success";
		message = "User is deleted!"						
}
redirectAttributes.addFlashAttribute("css", typeCss);
redirectAttributes.addFlashAttribute("msg", message);
return "redirect:/admin/users";

Reply:
Dạ, em hiểu rồi ạ, thank chị <3